webroot/src/components/CompactSelector/CompactSelector.ts

```diff
@@ -0,0 +1,166 @@
+//
+//  CompactSelector.ts
+//  CompactConnect
+//
+//  Created by InspiringApps on 10/2/2024.
+//
+
+import { compacts as compactsConfig } from '@/app.config';
+import {
+    Component,
+    mixins,
+    Prop,
+    Watch,
+    toNative
+} from 'vue-facing-decorator';
+import { reactive, computed, ComputedRef } from 'vue';
+import { RouteRecordName } from 'vue-router';
+import MixinForm from '@components/Forms/_mixins/form.mixin';
+import InputSelect from '@components/Forms/InputSelect/InputSelect.vue';
+import { FormInput } from '@models/FormInput/FormInput.model';
+import { Compact, CompactSerializer, CompactType } from '@models/Compact/Compact.model';
+import { User, CompactPermission } from '@models/User/User.model';
+
+interface CompactOption {
+    value: string | number;
+    name: string | ComputedRef<string>;
+}
+
+@Component({
+    name: 'CompactSelector',
+    components: {
+        InputSelect
+    }
+})
+class CompactSelector extends mixins(MixinForm) {
+    @Prop({ default: false }) isPermissionBased?: boolean;
+    @Prop({ default: false }) hideIfNotMultiple?: boolean;
+
+    //
+    // Lifecycle
+    //
+    created() {
+        this.init();
+    }
+
+    //
+    // Computed
+    //
+    get userStore() {
+        return this.$store.state.user;
+    }
+
+    get user(): User | null {
+        return this.userStore.model;
+    }
+
+    get currentCompact(): Compact | null {
+        return this.userStore.currentCompact;
+    }
+
+    get userPermissions(): Array<CompactPermission> {
+        return this.user?.permissions || [];
+    }
+
+    get allCompacts(): Array<Compact> {
+        const compactTypes = Object.keys(compactsConfig) as Array<CompactType>;
+        const compacts = compactTypes.map((compactType) => new Compact({ type: compactType }));
+
+        return compacts;
+    }
+
+    get compactOptions(): Array<CompactOption> {
+        const options: Array<CompactOption> = [];
+
+        if (this.isPermissionBased) {
+            this.userPermissions.forEach((permission: CompactPermission) => {
+                const { compact } = permission;
+
+                options.push({ value: (compact.type as unknown as string), name: compact.name() });
+            });
+        } else {
+            this.allCompacts.forEach((compact) => {
+                options.push({ value: (compact.type as unknown as string), name: compact.name() });
+            });
+        }
+
+        return options;
+    }
+
+    get shouldDisplay(): boolean {
+        const { compactOptions, hideIfNotMultiple } = this;
+        const optionsLength = compactOptions.length;
+
+        return !optionsLength || !(optionsLength === 1 && hideIfNotMultiple);
```

Comment:
This took me a few seconds to parse...Do you need the first clause? I think this can be simplified to:

```suggestion
        return !(optionsLength === 1 && hideIfNotMultiple);
```

or even

```suggestion
        return optionsLength !== 1 || !hideIfNotMultiple);
```

Reply:
I think it can be simplified. I didn't have a design state for edge cases in Figma, but I think we can keep it simpler for now.